### PR TITLE
chore: remove all mutexes and store in subscription struct

### PR DESCRIPTION
**Pull request by im-adithya**
This removes the need of mutexes as the suibscription struct is carried everywhere, so it's best to add the `relaySubscription` to the subscription itself instead of adding those to `svc.subscriptions`

### CODE REVIEW

**Comment by im-adithya** [comment]
⚠️ THIS WILL NOT WORK WITH CONTINUOUS SUBSCRIPTIONS

File: internal/nostr/nostr.go [hunk 1]
```diff
@@ -685,15 +681,11 @@ func (svc *Service) StopSubscriptionHandler(c echo.Context) error {
...
-	if exists {
-		sub.Unsub()
-		delete(svc.subscriptions, subscription.Uuid)
+	if subscription.RelaySubscription != nil {
+		subscription.RelaySubscription.Unsub()
 	}
-	svc.subscriptionsMutex.Unlock()
 
-	if (!exists && !subscription.Open) {
+	if (!subscription.Open) {
```
[the remark this answers is not in the document]
Comment (rolznz) [reply internal/nostr/nostr.go:688]: is it because it only applies for .e.g subscribing to notifications?

[the remark this answers is not in the document]
Comment (im-adithya) [reply internal/nostr/nostr.go:688]: Yes ✅

File: internal/nostr/nostr.go [hunk 2]
```diff
@@ -804,7 +792,7 @@ func (svc *Service) publishRequestEvent(ctx context.Context, subscription *Subsc
 			"client_pubkey":    clientPubkey,
 		}).Error("Failed to publish to relay")
 		subscription.RequestEvent.State = REQUEST_EVENT_PUBLISH_FAILED
-		sub.Unsub()
+		relaySubscription.Unsub()
```
Comment (rolznz) [inline internal/nostr/nostr.go:795]: should stopSubscription be called here? or not?

**Review by rolznz — APPROVED** [review]
utACK, good simplification

File: internal/nostr/nostr.go (hunk 2 above)
Comment (im-adithya) [reply internal/nostr/nostr.go:795]: Did that because it's directly available, but good to keep it consistent :+1: will change to stopSubscription

**Pull request merged**